todo_app/app.py

```diff
@@ -107,152 +202,168 @@ def get_index():
   
     @app.route('/new', methods=['GET'])   # New Task
     @login_required
+    @usermanager.hasWritePermission
     def getnew_post():
-        if (app.config['LOGIN_DISABLED']) or (usermanager.IsRoleAdmin(current_user.id)) or (usermanager.IsRoleWriter(current_user.id)):
-            return render_template('new_task.html')
-        return render_template("access_error.html",error="insufficient privileges!")
-
+        return render_template('new_task.html')
 
     @app.route('/home', methods=['POST'])   # New Task
     @login_required
+    @usermanager.hasWritePermission
     def post_index():
-        if (app.config['LOGIN_DISABLED']) or (usermanager.IsRoleAdmin(current_user.id)) or (usermanager.IsRoleWriter(current_user.id)):
-            response = todo.create_task(
-                name = request.form['title'],
-                due = request.form['duedate'],
-                desc = request.form['descarea']
-            )        
-            if (str(response) != ""):
-                return redirect('/home')
-            else:
-                return render_template("error.html",error="failed to create task!")
-        
-        return render_template("access_error.html",error="insufficient privileges!")
+        response = todo.create_task(
+            name = request.form['title'],
+            due = request.form['duedate'],
+            desc = request.form['descarea']
+        )        
+        if response is not None:
+            if current_app.config["LOGIN_DISABLED"]==False:
```

Comment:
Why are you only logging this info when LOGIN_DISABLED is false - is this to avoid tests logging? Rather than app logic, I'd be inclined to use some combination of:
* Control the log level appropriately
* Send your logs to slightly different places, or use different tags so that you can filter them